compiler/resolution/functionResolution.cpp

```diff
@@ -1517,12 +1517,28 @@ bool canCoerceAsSubtype(Type*     actualType,
       return true;
   }
 
+  // coerce c_ptr to c_ptrConst
+  if (actualType->symbol->hasFlag(FLAG_C_PTR_CLASS) &&
+      !startsWith(actualType->name(), "c_ptrConst") &&
+      formalType->symbol->hasFlag(FLAG_C_PTR_CLASS) &&
+      startsWith(formalType->name(), "c_ptrConst")) {
+    // check element types match
+    Type* actualElt = getDataClassType(actualType->symbol)->typeInfo();
+    Type* formalElt = getDataClassType(formalType->symbol)->typeInfo();
+    if (actualElt && formalElt && actualElt == formalElt) {
```

Comment:
My eyes (potentially some compilers, too, I hope) would be happier if you parenthesize `actualElt == formalElt`

Reply:
Agreed, and I will also change this in the code I copied it from below (for coercion of c_array to c_ptr)